app/flame_chart/flame_chart_model.ts

```diff
@@ -0,0 +1,103 @@
+import { mod } from "../util/math";
+import colors from "./colors";
+import { buildThreadTimelines, ThreadEvent, TraceEvent } from "./profile_model";
+import {
+  BLOCK_HEIGHT,
+  BLOCK_VERTICAL_GAP,
+  SECTION_LABEL_HEIGHT,
+  SECTION_PADDING_BOTTOM,
+  SECTION_PADDING_TOP,
+} from "./style_constants";
+
+export type FlameChartModel = {
+  sections: SectionDecorationModel[];
+  blocks: BlockModel[];
+};
+
+export type SectionDecorationModel = {
+  name: string;
+  y: number;
+  height: number;
+};
+
+export type BlockModel = {
+  rectProps: {
+    x: number;
+    y: number;
+    width: number;
+    fill: string;
+  };
+  event: ThreadEvent;
+};
+
+/**
+ * Returns a random color for the given ID.
+ *
+ * Calls for the same ID will return the same color.
+ *
+ * All colors returned have the same approximate perceived brightness
+ * to avoid issues with color contrast.
+ */
+const getColor = (id: string) => colors[mod(hash(id), colors.length)];
```

Comment:
wdyt about colors[Math.abs(hash(id)) % colors.length], or throwing abs in hash so it doesn't return negative numbers

feels better than writing your own mod function

Reply:
Done